ra-optimisation/search-tree-rao/src/main/java/com/powsybl/openrao/searchtreerao/result/impl/PreventiveAndCurativesRaoResultImpl.java

```diff
@@ -585,6 +586,19 @@ public void setOptimizationStepsExecuted(OptimizationStepsExecuted optimizationS
         }
     }
 
+    @Override
+    public boolean isSecure(Instant optimizedInstant, PhysicalParameter... u) {
+        if (ComputationStatus.FAILURE.equals(getComputationStatus())) {
+            return false;
+        }
+        return getCost(optimizedInstant) < 0;
```

Comment:
virtual costs may include MNEC costs, LoopFlow costs, etc that do not count in security status. If you want to check divergence, check the computation status map first like you did with computation status
```suggestion
        return getFunctionalCost(optimizedInstant) < 0;
```